src/ClimaSeaIce.jl

```diff
@@ -1,142 +1,37 @@
-
-    return LinearLiquidus(convert(FT, freshwater_melting_temperature),
-                          convert(FT, slope))
-end
-
-@inline function melting_temperature(liquidus::LinearLiquidus, salinity)
-    return liquidus.freshwater_melting_temperature - liquidus.slope * salinity
-end
-
-struct PhaseTransitions{FT, L}
-    ice_density :: FT
-    ice_heat_capacity :: FT
-    liquid_density :: FT
-    liquid_heat_capacity :: FT
-    reference_latent_heat :: FT
-    reference_temperature :: FT
-    liquidus :: L
-end
-
-"""
-    PhaseTransitions(FT=Float64,
-                     ice_density           = 917,   # kg m⁻³
-                     ice_heat_capacity     = 2000,  # J / (kg ᵒC)
-                     liquid_density        = 999.8, # kg m⁻³
-                     liquid_heat_capacity  = 4186,  # J / (kg ᵒC)
-                     reference_latent_heat = 334e3  # J kg⁻³
-                     liquidus = LinearLiquidus(FT)) # default assumes psu, ᵒC
-
-Return a representation of transitions between the solid and liquid phases
-of salty water: in other words, the freezing and melting of sea ice.
-
-The latent heat of fusion ``ℒ(T)`` (more simply just "latent heat") is
-a function of temperature ``T`` via
-
-```math
-ρᵢ ℒ(T) = ρᵢ ℒ₀ + (ρ_ℓ c_ℓ - ρᵢ cᵢ) (T - T₀)    
-```
-
-where ``ρᵢ`` is the `ice_density`, ``ρ_ℓ`` is the liquid density,
-``cᵢ`` is the heat capacity of ice, and ``c_ℓ`` is the heat capacity of
-liquid, and ``T₀`` is a reference temperature, all of which are assumed constant.
-
-The default `liquidus` assumes that salinity has practical salinity units (psu)
-and that temperature is degrees Celsius.
-"""
-@inline function PhaseTransitions(FT=Float64;
-                                  ice_density           = 917,    # kg m⁻³
-                                  ice_heat_capacity     = 2000,   # J / (kg ᵒC)
-                                  liquid_density        = 999.8,  # kg m⁻³
-                                  liquid_heat_capacity  = 4186,   # J / (kg ᵒC)
-                                  reference_latent_heat = 334e3,  # J kg⁻³
-                                  reference_temperature = 0,      # ᵒC
-                                  liquidus = LinearLiquidus(FT))
-
-    return PhaseTransitions(convert(FT, ice_density),
-                            convert(FT, ice_heat_capacity),
-                            convert(FT, liquid_density),
-                            convert(FT, liquid_heat_capacity),
-                            convert(FT, reference_latent_heat),
-                            convert(FT, reference_temperature),
-                            liquidus)
-end
-
-@inline function latent_heat(thermo::PhaseTransitions, T)
-    T₀ = thermo.reference_temperature    
-    ℒ₀ = thermo.reference_latent_heat
-    ρᵢ = thermo.ice_density
-    ρℓ = thermo.liquid_density
-    cᵢ = thermo.ice_heat_capacity
-    cℓ = thermo.liquid_heat_capacity
-
-    return ρℓ * ℒ₀ + (ρℓ * cℓ - ρᵢ * cᵢ) * (T - T₀)
-end
```

Comment:
What do all these changes have to do with putting the thermodynamics in a model? Why can't we proceed incrementally? This PR changes almost every file in the package.

Reply:
I moved all this code, that relates to the thermodynamics, to the `SeaIceThermodynamics` module